internal/am/upload_transfer.go

```diff
@@ -113,3 +145,14 @@ func (a *UploadTransferActivity) Heartbeat(ctx context.Context, upload sftp.Asyn
 		}
 	}
 }
+
+func formatUploadError(err error) error {
```

Comment:
I think the name "uploadError" (or something similar) would be better because there is a functional difference between returning a `temporal.NewNonRetryableError(e)` and a standard (retryable) error.